angularSTEP/src/app/item-dialog/item-dialog.component.html

```diff
@@ -0,0 +1,32 @@
+<div *ngIf="add"> 
+  <h1 mat-dialog-title>Add Item to Shopping List</h1>
+<!-- <div mat-dialog-content>
+    {{data.value}}
+</div> -->
+  <p>How many times would you like to add "{{selectedItem}}" to your Shopping List?</p>
+  <mat-dialog-content [formGroup]="form">
+    <mat-form-field>
+      <input matInput type="number" min="1" formControlName="quantity" required>
+    </mat-form-field>
+  </mat-dialog-content>
+
+  <mat-dialog-actions>
+    <button mat-raised-button color="primary" (click)="cancel()">Cancel</button>
+    <button mat-raised-button color="primary" (click)="close()">Add</button>
+  </mat-dialog-actions>
+</div>
+
+<div *ngIf="remove">
+  <h1 mat-dialog-title>Remove Item from Shopping List</h1>
+  <p>How many times would you like to remove "{{selectedItem}}" from your Shopping List?</p>
+  <mat-dialog-content [formGroup]="form">
+    <mat-form-field>
+      <input #removeInput matInput type="number" min="1" max="{{max}}" formControlName="quantity" required>
+    </mat-form-field>
+  </mat-dialog-content>
+
+  <mat-dialog-actions>
+    <button mat-raised-button color="primary" (click)="cancel()">Cancel</button>
+    <button mat-raised-button color="primary" (click)="close()">Remove</button>
```

Comment:
Could this (click) callback be named 'remove' instead of 'close'?

Reply:
Since I only created one dialog component that is used for both adding and removing and that use the same line of code to close and send their result back to where the dialog opened, I used the same function for both. I could name change it to `save()` instead of `close()` or I can separate the two and have `add()` and `remove()`

Reply:
Splitting up for different functions sounds great! It helps convey a lot more information about exactly what should be changing within the way the function is called, rather than implicitly thorough its argument and other state